stability-compiler/src/main/kotlin/com/skydoves/compose/stability/compiler/lower/StabilityAnalyzerTransformer.kt

```
@@ -434,6 +439,29 @@ public class StabilityAnalyzerTransformer(
    * Matches K2 implementation logic.
    */
   private fun analyzeClassProperties(clazz: IrClass, fqName: String?): ParameterStability {
+    // Issue #31: Check if parent sealed class has @Immutable/@Stable
+    val parentHasStabilityAnnotation = clazz.superTypes.any { superType ->
+      val superClassSymbol = superType.classOrNull
+      if (superClassSymbol != null) {
+        val superClass = superClassSymbol.owner
+        // Check if superclass is sealed AND has stability annotation
+        val isSealed = try {
+          superClass.sealedSubclasses.isNotEmpty()
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)